src/Widgets/VpnMenuItem.vala

```diff
@@ -1,156 +1,97 @@
 /*
- * Copyright 2017-2020 elementary, Inc. (https://elementary.io)
- *
- * This program is free software: you can redistribute it and/or modify
- * it under the terms of the GNU Library General Public License as published by
- * the Free Software Foundation, either version 2.1 of the License, or
- * (at your option) any later version.
- *
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU Library General Public License for more details.
- *
- * You should have received a copy of the GNU Library General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- */
+* SPDX-License-Identifier: LGPL-2.1-or-later
+* SPDX-FileCopyrightText: 2017-2023 elementary, Inc. (https://elementary.io)
+*/
 
-public class Network.VpnMenuItem : Gtk.ListBoxRow {
-    public signal void user_action ();
+public class Network.VpnMenuItem : Gtk.FlowBoxChild {
+    public NM.RemoteConnection remote_connection { get; construct; }
 
-    public Gtk.RadioButton radio_button { get; private set; }
-    public Network.State vpn_state { get; set; default = Network.State.DISCONNECTED; }
-    public NM.RemoteConnection? connection { get; construct; }
-
-    public string id {
+    private NM.VpnConnection? _vpn_connection = null;
+    public NM.VpnConnection? vpn_connection {
         get {
-            return connection.get_id ();
+            return _vpn_connection;
         }
-    }
+        set {
+            if (value != null) {
+                _vpn_connection = value;
+                _vpn_connection.vpn_state_changed.connect (update_state);
```

Comment:
@danirabbit when `vpn_connection` is cleared, setter has responsibility to change view (update icon and so on). Maybe it would be better to have this responsibility when `vpn_connection` set, would it better to `update_state` immediately?

Reply:
Good call :)